litestar/routes/http.py

```diff
@@ -36,32 +23,20 @@ def __init__(
         self,
         *,
         path: str,
-        route_handlers: list[HTTPRouteHandler],
+        route_handlers: Iterable[HTTPRouteHandler],
     ) -> None:
         """Initialize ``HTTPRoute``.
 
         Args:
             path: The path for the route.
             route_handlers: A list of :class:`~.handlers.HTTPRouteHandler`.
         """
-        methods = list(chain.from_iterable([route_handler.http_methods for route_handler in route_handlers]))
-        if "OPTIONS" not in methods:
-            methods.append("OPTIONS")
-            options_handler = self.create_options_handler(path)
-            options_handler.owner = route_handlers[0].owner
-            route_handlers.append(options_handler)
-
-        self.route_handlers = route_handlers
-        self.route_handler_map: dict[Method, tuple[HTTPRouteHandler, KwargsModel]] = {}
+        super().__init__(path=path)
+        self.route_handler_map: dict[Method, HTTPRouteHandler] = self.create_handler_map(route_handlers)
+        self.route_handlers = tuple(self.route_handler_map.values())
+        self.methods = tuple(self.route_handler_map)
```

Comment:
These might as well be properties, but that requires implementing a `__deepcopy__` method, and I hope we can get rid of the copying entirely, so I'm going to leave this as-is for now.